### PR TITLE
String view type fixes

### PR DESCRIPTION
**Pull request by artemartey**

### CODE REVIEW

File: include/EASTL/string_view.h [hunk 1]
```diff
@@ -459,10 +459,10 @@ namespace eastl
 		size_t operator()(const string_view& x) const
 		{
 			string_view::const_iterator p = x.cbegin();
-			size_t result = 2166136261U; // We implement an FNV-like string hash.
+			uint32_t result = 2166136261U; // We implement an FNV-like string hash.
```
Comment (virt00l) [inline include/EASTL/string_view.h]: This is looks like it will work, altough I not verified it yet.

File: include/EASTL/string_view.h [hunk 2]
```diff
@@ -459,10 +459,10 @@ namespace eastl
...
 			while (p != x.cend())
```
Comment (virt00l) [inline include/EASTL/string_view.h]: I would consider to extract (i.e. to cache in local var) 'x.cend()' from loop (as it's actually loop invariant, i.e. is not changing within loop). Some compilers are probably smart enough to do it automatically, for others - it is need to be done manually

**Comment by artemartey** [comment]
all compilers I knew can make these optimizations, but maybe you're right.

**Comment by rparolin** [comment]
Looks good.  Thanks!

**Pull request merged**